src/sentry/api/endpoints/organization_member/__init__.py

```diff
@@ -32,9 +32,18 @@ def get_allowed_roles(
 
     allowed_roles = []
     if can_admin and not is_active_superuser(request):
-        acting_member = member or OrganizationMember.objects.get(
-            user=request.user, organization=organization
-        )
+        if member:
+            acting_member = member
+        else:
+            try:
+                acting_member = OrganizationMember.objects.get(
+                    user=request.user, organization=organization
+                )
+            except OrganizationMember.DoesNotExist:
+                # This can happen if the request was authorized by an app integration
+                # token whose proxy user does not have an OrganizationMember object.
+                return can_admin, allowed_roles
```

Comment:
for my own understanding, would this return `True` and `[]`?  and if it does, would the API have the correct role to function or would the API be restricted?

Reply:
> for my own understanding, would this return `True` and `[]`?

Yes.

> and if it does, the API would fail as it does have the correct roles

It would "fail" in the sense of providing a graceful 403 error. Our system currently lacks any way to authorize an app integration token to assign any particular set of roles. So `[]` as allowed roles is the "correct" answer but not the "best" answer. The best answer would be blocked by further back-end changes; this PR only patches over the 500 response.